src/Core/Addon/AddonManagerHelper.php

```diff
@@ -0,0 +1,306 @@
+	{
+		return $this->addonPath;
+	}
+
+	/**
+	 * Returns the list of available addons.
+	 *
+	 * This list is made from scanning the addon/ folder.
+	 * Unsupported addons are excluded unless they already are enabled or system.show_unsupported_addon is set.
+	 *
+	 * @return string[]
+	 */
+	public function getAvailableAddons(): array
+	{
+		$dirs = scandir($this->getAddonPath());
+
+		if (!is_array($dirs)) {
+			return [];
+		}
+
+		$files = [];
+
+		foreach ($dirs as $dirname) {
+			// ignore hidden files and folders
+			// @TODO: Replace with str_starts_with() when PHP 8.0 is the minimum version
+			if (\strncmp($dirname, '.', 1) === 0) {
+				continue;
+			}
+
+			if (!is_dir($this->getAddonPath() . '/' . $dirname)) {
+				continue;
+			}
+
+			$files[] = $dirname;
+		}
+
+		$addons = [];
+
+		foreach ($files as $addonId) {
+			$addonInfo = $this->getAddonInfo($addonId);
+
+			if (
+				$this->config->get('system', 'show_unsupported_addons')
+				|| strtolower($addonInfo->getStatus()) !== 'unsupported'
+				|| $this->isAddonEnabled($addonId)
+			) {
+				$addons[] = $addonId;
+			}
+		}
+
+		return $addons;
+	}
+
+	/**
+	 * Installs an addon.
+	 *
+	 * @param string $addonId name of the addon
+	 *
+	 * @return bool true on success or false on failure
+	 */
+	public function installAddon(string $addonId): bool
+	{
+		$addonId = Strings::sanitizeFilePathItem($addonId);
+
+		$addon_file_path = $this->getAddonPath() . '/' . $addonId . '/' . $addonId . '.php';
+
+		// silently fail if addon was removed or if $addonId is funky
+		if (!file_exists($addon_file_path)) {
+			return false;
+		}
+
+		$this->logger->debug("Addon {addon}: {action}", ['action' => 'install', 'addon' => $addonId]);
+
+		$timestamp = @filemtime($addon_file_path);
+
+		@include_once($addon_file_path);
+
+		if (function_exists($addonId . '_install')) {
+			$func = $addonId . '_install';
+			$func();
+		}
+
+		$this->config->set('addons', $addonId, [
+			'last_update' => $timestamp,
+			'admin'       => function_exists($addonId . '_addon_admin'),
+		]);
+
+		if (!$this->isAddonEnabled($addonId)) {
+			$this->addons[] = $addonId;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Uninstalls an addon.
+	 *
+	 * @param string $addonId name of the addon
+	 */
+	public function uninstallAddon(string $addonId): void
+	{
+		$addonId = Strings::sanitizeFilePathItem($addonId);
+
+		$this->logger->debug("Addon {addon}: {action}", ['action' => 'uninstall', 'addon' => $addonId]);
+		$this->config->delete('addons', $addonId);
+
+		$addon_file_path = $this->getAddonPath() . '/' . $addonId . '/' . $addonId . '.php';
+
+		@include_once($addon_file_path);
+
+		if (function_exists($addonId . '_uninstall')) {
+			$func = $addonId . '_uninstall';
+			$func();
+		}
+
+		// Remove registered hooks for the addon
+		// Handles both relative and absolute file paths
+		$condition = ['`file` LIKE ?', "%/$addonId/$addonId.php"];
+
+		$result = $this->database->delete('hook', $condition);
+
+		if ($result) {
```

Comment:
Are there any caveats when checking that way?

Reply:
That's the code from `Hook::delete()`, see https://github.com/friendica/friendica/blob/2024.12/src/Core/Hook.php#L249-L253

Because `Database::delete()` returns bool this is safe to use, but I can also change it to `$result === true`.